### PR TITLE
ROX-25221: Remove Lerna from UI build process

### PR DESCRIPTION
**Pull request by dvail**
### Description

Removes Lerna from the UI build process, and de-monorepos the `ui/` monorepo.

Why?

1. We currently have monorepo tooling and a monorepo structure under `ui/` that manages only a single package
2. This tooling and structure complicates the codebase, and more pressingly, complicates integration with Konflux
3. There is no intention in the near future to once again divide the UI code into multiple packages under a monorepo structure
4. If there _was_, we would likely structure the project differently anyway

### Core process

1. The `yarn.lock` file was modified by removing top level dev deps with the following commands:
- `yarn remove lerna`
- `yarn remove chalk-cli`
2. The `resolutions` section of `ui/package.json` was moved to `ui/apps/platform/package.json`
3. The `workspaces` section of `ui/package.json` was removed
4. `yarn.lock` was moved into `ui/apps/platform`
5. (!) All node_modules were removed, and `yarn install` was run from `apps/platform`. **No changes occurred in the yarn.lock as a result of the install.**
6. All package.json scripts in the top level package.json that used lerna to stream to all child packages were updated to instead traverse down a directory and run `cd apps/platform; yarn <command>` instead.

### Future follow ups
1. Delete `ui/package.json` and combine scripts with `ui/apps/platform/package.json`

### User-facing documentation

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

- [x] contributed **no automated tests**

The CI build process is essentially the automated test for this change.

#### How I validated my change

Manually verify that all top level Make targets related to UI complete successfully:
```sh
make ui-lint && \
make ui-build && \
make ui-test && \
make ui-component-tests && \
make generate-junit-reports
```

Manually verify that all `ui` level Make targets related to UI complete successfully:
```sh
make clean && \
make all && \
make test-e2e && \
make start
```

Manually verify that dependency cache is created and detected correctly in GHA https://github.com/stackrox/stackrox/actions/runs/9894326245/job/27338197817?pr=11928#step:5:29 

Manually verify that the `test-results` directory is created under `ui/apps/platform` - both in this branch and on `master`.

Delete `node_modules`, then `yarn` in `apps/platform`. For each dependency under resolutions:
`grep version <pkg_dir>/package.json` and compare to the declared version in resolutions.

Manually verify that junit output is detected and read during CI jobs.
- [x] unit tests
- [x] component tests
- [x] e2e tests

`yarn start` in `apps/platform` and verify that the front end is served correctly.

Finally, pull the image from this PR and run it on cluster to verify that the build image was created successfully.

### CODE REVIEW

File: ui/apps/platform/package.json [hunk 1]
```diff
@@ -171,6 +171,18 @@
...
+        "@typescript-eslint/eslint-plugin": "^6.21.0",
+        "@typescript-eslint/parser": "^6.21.0",
+        "autoprefixer": "10.4.5",
+        "babel-jest": "^29.7.0",
+        "eslint-plugin-jest": "^27.6.0",
+        "eslint-plugin-testing-library": "^6.2.0",
+        "jest": "^29.7.0",
+        "jest-resolve": "^29.7.0",
+        "jest-watch-typeahead": "^2.2.2"
+    },
```
Comment (dvail) [inline ui/apps/platform/package.json:185]: TODO - Need to do a manual test to ensure resolutions are behaving as expected.

File: ui/Makefile [hunk 2]
```diff
@@ -40,12 +40,12 @@ start: deps
 .PHONY: test
 test: deps $(SOURCES)
 	@echo "+ $@"
-	cd apps/platform; yarn test
+	yarn test
```
Comment (dvail) [inline ui/Makefile:43]: The top level `yarn test` will delegate to the `yarn test` in `apps/platform`, and calling from the top level lets us use the `CI=true` variable without needing to modify the lower level package.json.

In a future cleanup, we will need to do this when the top level package.json is removed.

**Review by pedrottimark — APPROVED** [review]

**Review by stehessel — APPROVED** [review]
not sure why Merlin got tagged, but from my limited knowledge of npm and yarn it look good.

File: ui/Makefile [hunk 3]
```diff
@@ -13,8 +13,8 @@ PACKAGE_JSON_FILES := $(shell find . -type d \( -name node_modules \) -prune -fa
 ROX_PRODUCT_BRANDING ?= $(shell $(MAKE) --quiet --no-print-directory -C .. product-branding)
 export REACT_APP_ROX_PRODUCT_BRANDING := $(ROX_PRODUCT_BRANDING)
 
-deps: yarn.lock $(PACKAGE_JSON_FILES)
-	yarn install --frozen-lockfile --prefer-offline --network-timeout=$(YARN_NETWORK_TIMEOUT)
+deps: apps/platform/yarn.lock $(PACKAGE_JSON_FILES)
+	cd apps/platform; yarn install --frozen-lockfile --prefer-offline --network-timeout=$(YARN_NETWORK_TIMEOUT)
```
[the remark this answers is not in the document]
Comment (dvail) [reply ui/Makefile:17]: Ah yep, I didn't know about this but I like it a lot better. I'll change occurrences of `cd` to use this instead. Thanks for the review!

**Review by janisz — APPROVED** [review]
Looked at .github changes and they look good

**Pull request merged**